velox/exec/SortBuffer.cpp

```diff
@@ -285,11 +285,13 @@ void SortBuffer::spillOutput() {
 }
 
 void SortBuffer::prepareOutput(uint32_t maxOutputRows) {
-  VELOX_CHECK_GT(maxOutputRows, 0);
+  VELOX_CHECK_GT(static_cast<vector_size_t>(maxOutputRows), 0);
   VELOX_CHECK_GT(numInputRows_, numOutputRows_);
-
+  const uint64_t batchSizeMaybe = numInputRows_ - numOutputRows_;
   const vector_size_t batchSize =
-      std::min<vector_size_t>(numInputRows_ - numOutputRows_, maxOutputRows);
+      batchSizeMaybe > std::numeric_limits<vector_size_t>::max()
```

Comment:
maxOutputRows is uint32_t, so can still overflow when cast to vector_size_t, no?

If it is hard to change the signature of this method, perhaps we can add a check that maxOutputRows doesn't exceed the max for  vector_size_t